opaque_keys/edx/locator.py

```diff
@@ -1330,3 +1330,72 @@ def to_deprecated_list_repr(self):
 
 # Register AssetLocator as the deprecated fallback for AssetKey
 AssetKey.set_deprecated_fallback(AssetLocator)
+
+
+class AggregateCourseLocator(AggregateCourseKey):    # pylint: disable=abstract-method
+    """
+    An AggregateCourseKey implementation class.
+    """
+    CANONICAL_NAMESPACE = 'aggregate-course'
+    KEY_FIELDS = ('org', 'course')
+    __slots__ = KEY_FIELDS
+
+    URL_RE_SOURCE = r'^(?P<org>{ALLOWED_ID_CHARS}+)\+(?P<course>{ALLOWED_ID_CHARS}+)$'.format(
+        ALLOWED_ID_CHARS=Locator.ALLOWED_ID_CHARS
+    )
+
+    URL_RE = re.compile(URL_RE_SOURCE)
+
+    def __init__(self, org=None, course=None, **kwargs):
+        """
+        Construct a AggregateCourseLocator.
+
+        Arguments:
+            org (string): Organization identifier for the course
+            course (string): Course name
```

Comment:
My understanding is we call the middle part of key as course or course name `organization+course_name+course_run` and all together are considered as a unique identifier for a course. I can change it if its confusing.